pandas/core/groupby/groupby.py

```diff
@@ -1420,8 +1420,12 @@ def size(self):
         """
         result = self.grouper.size()
 
+        # GH28330 preserve subclassed Series/DataFrames through calls
         if isinstance(self.obj, Series):
+            result = self.obj._constructor(result)
```

Comment:
try using assert here as well

Reply:
Thanks -- this sort of assertion worked to avoid unnecessary calls to the constructor